diff --git a/www/panel/NodeInfo.js b/www/panel/NodeInfo.js
index 72f97c7c..b8b2c4cc 100644
--- a/www/panel/NodeInfo.js
+++ b/www/panel/NodeInfo.js
@@ -151,7 +151,7 @@ Ext.define('PBS.NodeInfoPanel', {
 		    return data.kversion;
 		}
 		let kernel = data['current-kernel'];
-		let buildDate = kernel.version.match(/\((.+)\)\s*$/)[1] ?? 'unknown';
+		let buildDate = kernel.version;
 		return `${kernel.sysname} ${kernel.release} (${buildDate})`;
 	    },
 	    value: '',
