diff --git a/www/NavigationTree.js b/www/NavigationTree.js
index 576d05ab..7ba9955a 100644
--- a/www/NavigationTree.js
+++ b/www/NavigationTree.js
@@ -77,12 +77,6 @@ Ext.define('PBS.store.NavigationStore', {
 		expanded: true,
 		leaf: false,
 		children: [
-		    {
-			text: gettext('Shell'),
-			iconCls: 'fa fa-terminal',
-			path: 'pbsXtermJsConsole',
-			leaf: true,
-		    },
 		    {
 			text: gettext('Storage / Disks'),
 			iconCls: 'fa fa-hdd-o',
