diff --git a/termproxy/debian/rules b/termproxy/debian/rules
index 6049291..cd9eabb 100755
--- a/termproxy/debian/rules
+++ b/termproxy/debian/rules
@@ -23,5 +23,5 @@ override_dh_auto_build:
 
 override_dh_auto_configure:
 	    die "ERROR: d/changelog <-> Cargo.toml version mismatch: $$v_cargo != $$v_deb\n" if $$v_cargo ne $$v_deb; exit(0); }' Cargo.toml
-	$(CARGO) prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
+	#$(CARGO) prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
 	dh_auto_configure
