From 7f5d351e3cdd343f00adb09df8acfcdaeec3012d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Trzci=C5=84ski?= <ayufan@ayufan.eu>
Date: Sat, 14 Jan 2023 10:44:37 +0100
Subject: [PATCH] Remove libproxmox-acme-perl

---
 debian/control | 1 -
 1 file changed, 1 deletion(-)

diff --git a/debian/control b/debian/control
index 232a0e4..743a5b5 100644
--- a/debian/control
+++ b/debian/control
@@ -33,7 +33,6 @@ Depends: libclone-perl,
          libnet-dbus-perl,
          libnet-ip-perl,
          libnetaddr-ip-perl,
-         libproxmox-acme-perl,
          libproxmox-rs-perl,
          libstring-shellquote-perl,
          libtimedate-perl,
-- 
2.34.1

