diff --git a/proxmox-rest-server/src/lib.rs b/proxmox-rest-server/src/lib.rs
index 03942519..01e4950b 100644
--- a/proxmox-rest-server/src/lib.rs
+++ b/proxmox-rest-server/src/lib.rs
@@ -101,7 +101,7 @@ pub fn read_pid(pid_fn: &str) -> Result<i32, Error> {
 pub fn ctrl_sock_from_pid(pid: i32) -> String {
     // Note: The control socket always uses @/run/proxmox-backup/ as prefix
     // for historc reason.
-    format!("\0{}/control-{}.sock", "/run/proxmox-backup", pid)
+    format!("{}/control-{}.sock", "/run/proxmox-backup", pid)
 }
 
 /// Returns the control socket path for this server.
