commit 1b425b8fd31147524f81c549ccbf0318a75f1c9e
Author: Kamil Trzciński <ayufan@ayufan.eu>
Date:   Sat May 1 14:40:06 2021 +0200

    Support local compile

diff --git a/Cargo.toml b/Cargo.toml
index 2804632c..2854259b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -105,7 +105,7 @@ pbs-tools = { path = "pbs-tools" }
 
 # Local path overrides
 # NOTE: You must run `cargo update` after changing this for it to take effect!
-[patch.crates-io]
+#[patch.crates-io]
 #proxmox-acme-rs = { path = "../proxmox-acme-rs" }
 #proxmox-apt = { path = "../proxmox-apt" }
 #proxmox-async = { path = "../proxmox/proxmox-async" }
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -134,3 +134,31 @@ pbs-tape = { path = "pbs-tape" }
 [features]
 default = []
 #valgrind = ["valgrind_request"]
+
+[patch.crates-io]
+proxmox-acme-rs = { path = "../proxmox-acme-rs" }
+proxmox-apt = { path = "../proxmox-apt" }
+proxmox-async = { path = "../proxmox/proxmox-async" }
+proxmox-compression = { path = "../proxmox/proxmox-compression" }
+proxmox-borrow = { path = "../proxmox/proxmox-borrow" }
+proxmox-fuse = { path = "../proxmox-fuse" }
+proxmox-http = { path = "../proxmox/proxmox-http" }
+proxmox-io = { path = "../proxmox/proxmox-io" }
+proxmox-lang = { path = "../proxmox/proxmox-lang" }
+proxmox-openid = { path = "../proxmox-openid-rs" }
+proxmox-router = { path = "../proxmox/proxmox-router" }
+proxmox-schema = { path = "../proxmox/proxmox-schema" }
+proxmox-section-config = { path = "../proxmox/proxmox-section-config" }
+proxmox-shared-memory = { path = "../proxmox/proxmox-shared-memory" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-serde = { path = "../proxmox/proxmox-serde" }
+proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
+proxmox-time = { path = "../proxmox/proxmox-time" }
+proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
+pxar = { path = "../pxar" }
+proxmox-metrics = { path = "../proxmox/proxmox-metrics" }
+proxmox-rest-server = { path = "../proxmox/proxmox-rest-server" }
+proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
+proxmox-auth-api = { path = "../proxmox/proxmox-auth-api" }
+proxmox-sortable-macro = { path = "../proxmox/proxmox-sortable-macro" }
+proxmox-ldap = { path = "../proxmox/proxmox-ldap" }
