diff --git a/.cargo/config b/.cargo/config
deleted file mode 100644
index 3b5b6e4..0000000
--- a/.cargo/config
+++ /dev/null
@@ -1,5 +0,0 @@
-[source]
-[source.debian-packages]
-directory = "/usr/share/cargo/registry"
-[source.crates-io]
-replace-with = "debian-packages"
diff --git a/Cargo.toml b/Cargo.toml
index 2f3a3a6..5f21be1 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,3 +17,8 @@ ureq = { version = "2.4", default-features = false, features = [ "gzip" ] }
 proxmox-io = "1"
 proxmox-lang = "1.1"
 proxmox-sys = "0.4"
+
+[patch.crates-io]
+proxmox-io = { path = "../proxmox/proxmox-io" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-lang = { path = "../proxmox/proxmox-lang" }
diff --git a/debian/cargo-checksum.json b/debian/cargo-checksum.json
new file mode 100644
index 0000000..0967ef4
--- /dev/null
+++ b/debian/cargo-checksum.json
@@ -0,0 +1 @@
+{}
diff --git a/debian/cargo_config b/debian/cargo_config
new file mode 100644
index 0000000..38bd089
--- /dev/null
+++ b/debian/cargo_config
@@ -0,0 +1,2 @@
+[build]
+rustflags = []
diff --git a/debian/compat b/debian/compat
new file mode 100644
index 0000000..b4de394
--- /dev/null
+++ b/debian/compat
@@ -0,0 +1 @@
+11
diff --git a/debian/control b/debian/control
new file mode 100644
index 0000000..0a35e40
--- /dev/null
+++ b/debian/control
@@ -0,0 +1,33 @@
+Source: rust-termproxy
+Section: admin
+Priority: optional
+Build-Depends: debhelper (>= 12),
+ dh-cargo,
+ cargo:native,
+ rustc:native,
+ libstd-rust-dev
+Maintainer: Proxmox Support Team <support@proxmox.com>
+Standards-Version: 4.5.1
+Vcs-Git: git://git.proxmox.com/git/pve-xtermjs.git
+Vcs-Browser: https://git.proxmox.com/?p=pve-xtermjs.git;a=summary
+Homepage: https://www.proxmox.com
+Rules-Requires-Root: no
+
+Package: pve-xtermjs
+Architecture: any
+Multi-Arch: allowed
+Depends:
+ ${misc:Depends},
+ ${shlibs:Depends},
+ ${cargo:Depends}
+Recommends:
+ ${cargo:Recommends}
+Suggests:
+ ${cargo:Suggests}
+Provides:
+ ${cargo:Provides}
+Built-Using: ${cargo:Built-Using}
+XB-X-Cargo-Built-Using: ${cargo:X-Cargo-Built-Using}
+Description: HTML/JS Shell client for Proxmox projects
+ This package provides an xterm.js client/proxy for Proxmoxprojects'  Hosts,
+ Proxmox VE containers or Proxmox VE VM's serial-terminals.
diff --git a/debian/docs b/debian/docs
deleted file mode 100644
index 8696672..0000000
--- a/debian/docs
+++ /dev/null
@@ -1 +0,0 @@
-debian/SOURCE
diff --git a/debian/rules b/debian/rules
index 2b62edd..b34534f 100755
--- a/debian/rules
+++ b/debian/rules
@@ -3,8 +3,14 @@
 %:
 	dh $@ --buildsystem cargo
 
+override_dh_auto_configure:
+	mkdir -p debian/cargo_home/
+	cp -av debian/cargo_config debian/cargo_home/config
+
 override_dh_auto_build:
 	dh_auto_build
 	sed -e 's/@VERSION@/${VERSION}/' src/www/index.html.tpl.in > src/www/index.html.tpl
 	sed -e 's/@VERSION@/${VERSION}/' src/www/index.html.hbs.in > src/www/index.html.hbs
 	rm src/www/index.html.tpl.in src/www/index.html.hbs.in
+
+override_dh_auto_test:
