diff --git a/src/api2/backup/environment.rs b/src/api2/backup/environment.rs
index 9cfc5534..a237cb66 100644
--- a/src/api2/backup/environment.rs
+++ b/src/api2/backup/environment.rs
@@ -195,7 +195,7 @@ impl BackupEnvironment {
         if size < data.chunk_size {
             data.small_chunk_count += 1;
             if data.small_chunk_count > 1 {
-                bail!("fixed writer '{}' - detected multiple end chunks (chunk size too small)");
+                bail!("fixed writer '{}' - detected multiple end chunks (chunk size too small)", data.name);
             }
         }
 
diff --git a/src/tape/drive/mod.rs b/src/tape/drive/mod.rs
index a50a8a9a..41d8f5c6 100644
--- a/src/tape/drive/mod.rs
+++ b/src/tape/drive/mod.rs
@@ -260,7 +260,7 @@ pub fn media_changer(
                         None => Ok(None),
                     }
                 }
-                _ => bail!("unknown drive type '{}' - internal error"),
+                _ => bail!("unknown drive type '{}' - internal error", drive),
             }
         }
         None => {
@@ -308,7 +308,7 @@ pub fn open_drive(
                     let handle = open_lto_tape_drive(&tape)?;
                     Ok(Box::new(handle))
                 }
-                _ => bail!("unknown drive type '{}' - internal error"),
+                _ => bail!("unknown drive type '{}' - internal error", drive),
             }
         }
         None => {
@@ -498,7 +498,7 @@ pub fn request_and_load_media(
                         update_and_log_request_error(&mut last_error, request_error)?;
                     }
                 }
-                _ => bail!("drive type '{}' not implemented!"),
+                _ => bail!("drive type '{}' not implemented!", drive),
             }
         }
         None => {
@@ -587,7 +587,7 @@ fn tape_device_path(
                 "lto" => {
                     LtoTapeDrive::deserialize(config)?.path
                 }
-                _ => bail!("unknown drive type '{}' - internal error"),
+                _ => bail!("unknown drive type '{}' - internal error", drive),
             };
             Ok(path)
         }
