diff --git a/Cargo.toml b/Cargo.toml
index ffd72b2..623ee92 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,3 +14,4 @@ exclude = [ "build", "debian" ]
 curl = "0.4"
 clap = "2.33"
 proxmox = { path = "../proxmox/proxmox", default-features = false }
+proxmox-io = { path = "../proxmox/proxmox-io" }
diff --git a/src/main.rs b/src/main.rs
index 605a92e..4720b47 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -15,7 +15,7 @@ use mio::{Events, Interest, Poll, Token};
 
 use proxmox::sys::error::io_err_other;
 use proxmox::sys::linux::pty::{make_controlling_terminal, PTY};
-use proxmox::tools::byte_buffer::ByteBuffer;
+use proxmox_io::ByteBuffer;
 use proxmox::{io_bail, io_format_err};
 
 const MSG_TYPE_DATA: u8 = 0;
