diff --git a/Cargo.toml b/Cargo.toml
index 07203f40..509bb279 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -168,3 +168,5 @@ proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
 proxmox-time = { path = "../proxmox/proxmox-time" }
 proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
 proxmox-serde = { path = "../proxmox/proxmox-serde" }
+openssl = { git = "https://github.com/sfackler/rust-openssl", rev = "f218ecf0e87df2087412e3ab73a63b38d3f5e985" }
+openssl-sys = { git = "https://github.com/sfackler/rust-openssl", rev = "f218ecf0e87df2087412e3ab73a63b38d3f5e985" }
