diff --git a/src/Makefile b/src/Makefile
index ea8af24..d7f1351 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -2,7 +2,7 @@ PROGRAM=mini-journalreader
 SOURCES=mini-journalreader.c
 
 LIBS := libsystemd
-CFLAGS += -Werror -Wall -Wextra -Wl,-z,relro -g -O2
+CFLAGS += -Wall -Wextra -Wl,-z,relro -g -O2
 CFLAGS += -fstack-protector-strong -D_FORTIFY_SOURCE=2
 CFLAGS += $(shell pkg-config --cflags ${LIBS})
 LFLAGS += $(shell pkg-config --libs ${LIBS})
