diff --git a/debian/rules b/debian/rules
index 6049291..cd9eabb 100755
--- a/debian/rules
+++ b/debian/rules
@@ -23,5 +23,5 @@ override_dh_auto_build:
 
 override_dh_auto_configure:
 	rm -f Cargo.lock
-	/usr/share/cargo/bin/cargo prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
+	#/usr/share/cargo/bin/cargo prepare-debian $(CURDIR)/debian/cargo_registry --link-from-system
 	dh_auto_configure
