diff --git a/www/panel/NodeInfo.js b/www/panel/NodeInfo.js
--- a/www/panel/NodeInfo.js
+++ b/www/panel/NodeInfo.js
@@ -150,7 +150,7 @@ Ext.define('PBS.NodeInfoPanel', {
                     return data.kversion;
                 }
                 let kernel = data['current-kernel'];
-                let buildDate = kernel.version.match(/\((.+)\)\s*$/)?.[1] ?? 'unknown';
+                let buildDate = kernel.version;
                 return `${kernel.sysname} ${kernel.release} (${buildDate})`;
             },
             value: '',
