diff --git a/www/NavigationTree.js b/www/NavigationTree.js
index 649692c8..cec8d5a9 100644
--- a/www/NavigationTree.js
+++ b/www/NavigationTree.js
@@ -95,12 +95,6 @@ Ext.define('PBS.store.NavigationStore', {
                 expanded: true,
                 leaf: false,
                 children: [
-                    {
-                        text: gettext('Shell'),
-                        iconCls: 'fa fa-terminal',
-                        path: 'pbsXtermJsConsole',
-                        leaf: true,
-                    },
                     {
                         text: gettext('Storage / Disks'),
                         iconCls: 'fa fa-hdd-o',
