diff --git a/proxmox-sys/src/sys/linux/procfs/mod.rs b/proxmox-sys/src/sys/linux/procfs/mod.rs
index 5784e0e..089a798 100644
--- a/proxmox-sys/src/linux/procfs/mod.rs
+++ b/proxmox-sys/src/linux/procfs/mod.rs
@@ -455,8 +456,10 @@ pub fn read_meminfo() -> Result<ProcFsMemInfo, Error> {
 
     meminfo.swapused = meminfo.swaptotal - meminfo.swapfree;
 
-    let spages_line = file_read_firstline("/sys/kernel/mm/ksm/pages_sharing")?;
-    meminfo.memshared = spages_line.trim_end().parse::<u64>()? * 4096;
+    match file_read_firstline("/sys/kernel/mm/ksm/pages_sharing") {
+        Ok(spages_line) => meminfo.memshared = spages_line.trim_end().parse::<u64>()? * 4096,
+        Err(_) => meminfo.memshared = 0,
+    }
 
     Ok(meminfo)
 }
